src/main/java/gregtech/common/covers/detector/CoverDetectorBase.java

```diff
@@ -0,0 +1,91 @@
+package gregtech.common.covers.detector;
+
+import codechicken.lib.raytracer.CuboidRayTraceResult;
+import gregtech.api.cover.CoverBehavior;
+import gregtech.api.cover.ICoverable;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.text.TextComponentTranslation;
+
+public abstract class CoverDetectorBase extends CoverBehavior {
+    protected static final String NBT_KEY_IS_INVERTED = "isInverted";
+
+    private boolean isInverted;
+
+    public CoverDetectorBase(ICoverable coverHolder, EnumFacing attachedSide) {
+        super(coverHolder, attachedSide);
+        isInverted = false;
+    }
+
+    protected boolean isInverted() {
+        return this.isInverted;
+    }
+
+    protected void setInverted(boolean isInverted) {
+        this.isInverted = isInverted;
+    }
```

Comment:
Doesn't this also need the cover holder notifications that are present in `toggleInvertedWithNotification`?

Right now, `setInverted` is called from the GUI button presses, and the notifications only happen when the cover is screwdriver clicked. I would imagine that the notifications need to happen for all cases. Which if that is true, I don't see a point to the two different invert toggle methods.

Reply:
When `setInverted` is called from GUI, it will get automatically synced as GUI will handle it. That is why only screwdriving has synchronization.